app/controllers/notes_controller.rb

```diff
@@ -47,10 +46,10 @@ def note_params
   end
 
   def set_user_challenge
-    @user_challenge = current_user.user_challenges.find(params[:user_challenge_id])
+    @user_challenge = current_user.user_challenges.find_by(id: params[:user_challenge_id])
```

Comment:
`find_by` is better in certain cases because it returns `nil` when no record is found, instead of raising a `ActiveRecord::RecordNotFound` exception like `find` method would. This can be useful if you want your code to continue executing and handle the `nil` case in a graceful way, for example you may want to redirect the user to a "Not Found" page or show a flash message.

Another advantage of using `find_by` is that it's slightly more performant than `find` when you're expecting a large number of records to not be found, because `find` will raise an exception and will cause a rollback in the database, whereas `find_by` will not.